### PR TITLE
network_utils: use poll instead of select

### PR DESCRIPTION
**Pull request by fsimonfc**
select is limited to file descriptor numbers below FD_SETSIZE (1024), and calling FD_SET with a higher value can cause a buffer overflow. This can be problematic for applications using OpenDHT as a library.

### CODE REVIEW

**Comment by AmarOk1412** [comment]
Yeay!

File: src/network_utils.cpp [hunk 1]
```diff
@@ -235,23 +237,23 @@ UdpSocket::openSockets(const SockAddr& bind4, const SockAddr& bind6)
 
     running = true;
     rcv_thread = std::thread([this, stop_readfd, ls4=s4, ls6=s6]() mutable {
-        int selectFd = std::max({ls4, ls6, stop_readfd}) + 1;
+        struct pollfd fds[NUM_FDS];
+        for (int i = 0; i < NUM_FDS; i++)
+            fds[i].events = POLLIN;
+        int stop_readfd_index = 0, ls4_index = 1, ls6_index = 2;
```
Comment (aberaud) [inline src/network_utils.cpp:243]: constexpr unsigned or constexpr size_t

File: src/network_utils.cpp [hunk 2]
```diff
@@ -26,8 +26,10 @@
 #define close(x) closesocket(x)
 #define write(s, b, f) send(s, b, (int)strlen(b), 0)
 #else
-#include <sys/select.h>
 #include <fcntl.h>
+#include <poll.h>
```
Comment (aberaud) [inline src/network_utils.cpp:30]: what about Windows ?

**Review by aberaud — CHANGES_REQUESTED** [review]
looks mostly good

**Pull request merged**